src/theory/fp/theory_fp.cpp

```diff
@@ -593,23 +594,10 @@ void TheoryFp::preRegisterTerm(TNode node)
   }
   if (!options().fp.fpExp)
```

Comment:
Would it make sense to cache the option for faster access?

Reply:
I was just wondering as we do use it in the rewriter, but then not when we preregister terms, which is kinda weird. Maybe we should cache these for all theories where this pattern is used?

Reply:
Ah, it is different for the theory rewriters, since rewriters should not be options-dependent currently (since the rewritten form attribute is Node-level, not Env-level).

This is why we pass a flag to the FP theory rewriter instead of the options object, hence it is cached there.